fabric-gametest-api-v1/src/main/java/net/fabricmc/fabric/impl/gametest/FabricGameTestHelper.java

```diff
@@ -102,8 +102,10 @@ public static void runHeadlessServer(LevelStorage.Session session, ResourcePackM
 	public static void invokeTestMethod(TestContext testContext, Method method, Object testObject) {
 		try {
 			method.invoke(testObject, testContext);
-		} catch (IllegalAccessException | InvocationTargetException e) {
-			throw new RuntimeException("Failed to invoke test method (%s) in (%s)".formatted(method.getName(), method.getDeclaringClass().getCanonicalName()), e);
+		} catch (IllegalAccessException e) {
+			throw new RuntimeException("Failed to invoke test method (%s) in (%s) because %s".formatted(method.getName(), method.getDeclaringClass().getCanonicalName(), e.getMessage()), e);
+		} catch(InvocationTargetException e) {
+			throw new RuntimeException(e.getCause().getMessage(), e.getCause());
```

Comment:
I actually think if e.getCause() instanceof RuntimeException re, it should throw re directly

At least that's what Vanilla does